### PR TITLE
Add Mikrotik module

### PR DESCRIPTION
**Pull request by carlosedp**
Add RFC1213 MIB to be able to generate configs from OIDs 1.3.6.1.2.1.4.

Added Mikrotik Router generator module.

### CODE REVIEW

**Comment by brian-brazil** [comment]
We only add mibs for the sake of the example generator config as otherwise they're dead weight, so can you add a module there for the device type you're interested in?

**Comment by carlosedp** [comment]
Sure, added Mikrotik router module.

File: generator/generator.yml [hunk 1]
```diff
@@ -612,3 +612,42 @@ modules:
...
+      - source_indexes: [laIndex]
+        lookup: laNames
+        drop_source_indexes: true
+    overrides:
+      ifAlias:
+        ignore: true # Lookup metric
+      ifDescr:
+        ignore: true # Lookup metric
+      ifName:
+        ignore: true # Lookup metric
```
Comment (brian-brazil) [inline generator/generator.yml:651]: This is not looked up.

File: generator/generator.yml [hunk 2]
```diff
@@ -612,3 +612,42 @@ modules:
...
+      - laIndex
+      - sysUpTime
+      - sysDescr
+      - host
+      - mikrotik
+    lookups:
+      - source_indexes: [ifIndex]
+        lookup: ifAlias
+      - source_indexes: [ifIndex]
+        lookup: ifDescr
```
Comment (brian-brazil) [inline generator/generator.yml:631]: Do you really need both of these? Usually for a given device one is enough.

File: generator/generator.yml [hunk 3]
```diff
@@ -612,3 +612,42 @@ modules:
...
+      - source_indexes: [ifIndex]
+        lookup: 1.3.6.1.2.1.31.1.1.1.1
+      - source_indexes: [mtxrInterfaceStatsIndex]
+        lookup: ifAlias
+      - source_indexes: [mtxrInterfaceStatsIndex]
+        lookup: ifDescr
+      - source_indexes: [mtxrInterfaceStatsIndex]
+        lookup: 1.3.6.1.2.1.31.1.1.1.1
```
Comment (brian-brazil) [inline generator/generator.yml:639]: This doesn't seem right, what are you trying to do?

**Review by brian-brazil — COMMENTED** [review]
You'll need to regenerate snmp.yml.

File: generator/generator.yml (hunk 1 above)
Comment (carlosedp) [reply generator/generator.yml:639]: True, cleaned up.

File: generator/generator.yml (hunk 2 above)
Comment (carlosedp) [reply generator/generator.yml]: True, replaced by ifNames.

File: generator/generator.yml (hunk 3 above)
Comment (carlosedp) [reply generator/generator.yml]: True, replaced by ifNames.

**Comment by carlosedp** [comment]
I had to add unzip to the Dockerfile since it was failing when unzipping Kemp MIBs.

**Comment by brian-brazil** [comment]
Hmm, a whole pile of other things are changing including losing many interface types. Is this MIB you're adding pulling in some older stuff?

**Comment by carlosedp** [comment]
I believe RFC1213 was overriding some OIDs. Since I removed "ip" from my module, it's not required anymore.

File: snmp.yml [hunk 4]
```diff
@@ -15736,28 +19696,36 @@ paloalto_fw:
     type: DisplayString
     help: Zone name for the interface - 1.3.6.1.4.1.25461.2.1.2.3.10.1.1
     indexes:
-    - labelname: ifIndex
```
Comment (brian-brazil) [inline snmp.yml:15739]: Nothing here or below should be changing.

**Comment by carlosedp** [comment]
That's weird, I've only added the Mikrotik module. Maybe it wasn't generated properly before?

I'm using Docker to generate and download MIBs:

```sh
docker build -t snmp-generator .

docker run -ti \
  -v "${PWD}:/opt/" --entrypoint make snmp-generator mibs

docker run -ti \
  -v "${PWD}:/opt/" snmp-generator generate
```

**Comment by brian-brazil** [comment]
No, there's something weird going on. New indexes shouldn't be appearing on objects for paloalto_fw.

**Comment by SuperQ** [comment]
Please rebase in order to pick up some fixes to the generator.

**Comment by carlosedp** [comment]
Rebased. Apparently the weird generation for non-touched modules is gone.

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**